app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -37,18 +37,20 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
     ): View? {
         val view = inflater.inflate(R.layout.fragment_downloads, container, false)
 
-        val items = requireComponents.core.store.state.downloads.map {
-            DownloadItem(
-                it.value.id.toString(),
-                it.value.fileName,
-                it.value.filePath,
-                it.value.contentLength.toString(),
-                it.value.contentType,
-                it.value.status
-            )
-        }.filter {
-            it.status == DownloadState.Status.COMPLETED
-        }.filterNotExistsOnDisk()
+        val items = requireComponents.core.store.state.downloads.values
+            .sortedByDescending { it.createdTime } // sort from newest to oldest
```

Comment:
Could we add a test to cover the new feature? :)

Reply:
Sure! I'll try my hand at writing a test.

Reply:
I don't have much experience writing tests, but I thought I'd give it a shot.

I added `createdTime` field to DownloadItem and tried with the following code I wrote in DownloadControllerTest:

`@Test
    fun `downloads are sorted from newest to oldest`() {
        val testList = state.items

        val expectedList = testList.sortedByDescending { it.createdTime }

        assertEquals(testList, expectedList)
    }`

But that didn't work. I'm very new to this so maybe I'm not approaching this the right way, but I'd be happy to get some guidance! 😊

Reply:
@joc-a  [here](https://github.com/Amejia481/fenix/commit/4c2d5409e5b8433abbe20a50b8cb065876aeaad5) you can see an example. I would modify a bit `DownloadFragment` to make it testable.

Reply:
@Amejia481 Done, thanks so much!